relay-ourlogs/src/otel_to_sentry.rs

```diff
@@ -395,4 +442,264 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn parse_otel_log_without_span_and_trace_ids() {
+        // Test with completely missing traceId and spanId fields
+        let json = r#"{
+            "timeUnixNano": "1544712660300000000",
+            "observedTimeUnixNano": "1544712660300000000",
+            "severityNumber": 10,
+            "severityText": "Information",
+            "body": {
+                "stringValue": "Log without trace context"
+            },
+            "attributes": [
+                {
+                    "key": "test.attribute",
+                    "value": {
+                        "stringValue": "test value"
+                    }
+                }
+            ]
+        }"#;
+
+        let otel_log: OtelLogRecord = serde_json::from_str(json).unwrap();
+        let our_log = otel_to_sentry_log(otel_log, None, None);
+        let annotated_log: Annotated<OurLog> = Annotated::new(our_log);
+
+        assert_eq!(
+            get_path!(annotated_log.body),
+            Some(&Annotated::new("Log without trace context".into()))
+        );
+        assert_eq!(
+            annotated_log.value().unwrap().level.value().unwrap(),
+            &OurLogLevel::Info
+        );
+
+        let our_log = annotated_log.value().unwrap();
+        assert_trace_id_generated(our_log);
+        assert_span_id_empty(our_log);
+
+        assert_eq!(
+            annotated_log
+                .value()
+                .unwrap()
+                .attributes
+                .value()
+                .unwrap()
+                .get_value("test.attribute")
+                .unwrap()
+                .as_str(),
+            Some("test value")
+        );
+
+        // Note: trace_id will be a random UUID, so we only check that it exists and has the right metadata
+        let trace_id_value = annotated_log.value().unwrap().trace_id.value().unwrap();
+        assert!(!trace_id_value.to_string().is_empty());
+
+        let deterministic_log = create_deterministic_log_for_snapshot(&annotated_log);
```

Comment:
Insta supports redactions, they should work here: https://insta.rs/docs/redactions/